azure/scope/managedmachinepool.go

```diff
@@ -155,6 +155,11 @@ func buildAgentPoolSpec(managedControlPlane *infrav1exp.AzureManagedControlPlane
 		replicas = *machinePool.Spec.Replicas
 	}
 
+	subnetName := managedControlPlane.Spec.VirtualNetwork.Subnet.Name
```

Comment:
Do we have UT coverage for this L158 condition (when the subnet name is derived from the AzureManagedControlPlane's VNET subnet name)?